### DISCUSSION TITLE
Using Bepuphysics in Unity, is there a macro definition that can convert it to left-handed coordinate system?

### DISCUSSION THREAD

**Discussion by dannisliang** [bepu/bepuphysics1#28] [Q&A]
Using Bepuphysics in Unity, is there a macro definition that can convert it to left-handed coordinate system?  If not, which parts need to be modified to achieve it?

**Comment by RossNordby** [comment]
There is not, but the core functionality of the engine doesn't depend on handedness. If I remember correctly the things to watch out for are:
1. triangle sidedness (any one sided triangles will be flipped), 
2. helper functions like `Matrix.CreatePerspectiveFieldOfViewRH` which assume right handed coordinates (as the name implies), 
3. and some names like `Matrix3x3.Right`, `Matrix3x3.Up`, and `Matrix3x3.Forward` (corresponding to +X, +Y, and -Z respectively).

In other words, so long as you are consistent with the math, it should still work without changes.

Notably. bepuphysics2 dropped any coordinate system assumptions entirely since they aren't necessary for the physics to work.

**Reply by dannisliang** [reply]
Thank you for the quick response!

How to modify **CreateOrthographicRH()** and **CreateWorldRH()** Function to left-handed?

**Reply by RossNordby** [reply]
I can't confidently say 'yup that's right' to that many changes in sign without testing unfortunately :P

My personal preference would be to not make any changes to the existing code, and to simply be careful to maintain consistency where necessary. If I were to make any changes, I'd probably just *remove* handedness. For example, when it comes to vectors and matrices, you may want to just use `UnitX`/`UnitY`/`UnitZ` instead of directional names that imply handedness. `Matrix3x3` in v1 doesn't have those properties by default, but they're equivalent to the `Right`/`Up`/`Backward` properties as they're currently implemented. Just the raw values of the rows of the matrix.

Stuff like the projection helpers could likely be deleted and replaced by helpers from unity with your expected handedness.

**Reply by dannisliang** [reply]
Many thanks for your instruction~
